### PR TITLE
Improve performance of orphan cleanup slightly

### PR DESCRIPTION
**Pull request by dralley**
re: #6541
https://pulp.plan.io/issues/6581

### CODE REVIEW

**Comment by dralley** [comment]
Using 

* http://mirror.linux.duke.edu/pub/centos/8/BaseOS/x86_64/kickstart/
* https://dl.fedoraproject.org/pub/fedora/linux/releases/32/Everything/x86_64/os/

Before:

Resource manager killed by OOM, but according to Tanya it would take about 3.5 minutes on her laptop.

After:

> (pulp) [vagrant@pulp3-source-fedora32 devel]$ python3 orphan_cleanup_test.py                                            
> Sync time: 53 seconds (centos-kickstart)                                                                                                                        
> Sync time: 358 seconds (fedora32)                                                    
> Orphan cleanup time: 136 seconds

File: pulpcore/app/models/base.py [hunk 1]
```diff
@@ -98,7 +98,7 @@ class MasterModel(BaseModel, metaclass=MasterModelMeta):
 
     # This field must have a value when models are saved, and defaults to the value of
     # the TYPE attribute on the Model being saved (seen above).
-    pulp_type = models.TextField(null=False, default=None)
+    pulp_type = models.TextField(null=False, default=None, db_index=True)
```
Comment (bmbouter) [inline pulpcore/app/models/base.py:101]: :+1:

File: pulpcore/app/models/content.py [hunk 2]
```diff
@@ -429,7 +429,9 @@ class ContentArtifact(BaseModel, QueryMixin):
     Artifact is protected from deletion if it's present in a ContentArtifact relationship.
     """
 
-    artifact = models.ForeignKey(Artifact, on_delete=models.PROTECT, null=True)
+    artifact = models.ForeignKey(
+        Artifact, on_delete=models.PROTECT, null=True, related_name="content_memberships"
```
Comment (bmbouter) [inline pulpcore/app/models/content.py:433]: :+1:

**Review by bmbouter — APPROVED** [review]
This looks good to me. Thanks @dralley !

**Pull request merged**